packages/suite/src/middlewares/suite/buttonRequestMiddleware.ts

```diff
@@ -32,6 +32,25 @@ const buttonRequest =
             }
         }
 
+        // firmware bug https://github.com/trezor/trezor-firmware/issues/35
+        // ugly hack to make Cardano review modal work
+        // root cause of this bug is wrong button request ButtonRequest_Other from CardanoSignTx - should be ButtonRequest_SignTx
+        if (action.type === UI.REQUEST_BUTTON && action.payload.code === 'ButtonRequest_Other') {
```

Comment:
Ethereum is also guilty of using [`ButtonRequest_Other` ](https://github.com/trezor/trezor-suite/pull/7989/files#diff-0d74ef833a1277a56c3892aa64bb9683b997591d9a6acae3f58a0deafccd6baaR75), for the Gas step, I think.